src/storage.rs

```diff
@@ -12,6 +46,115 @@ impl Default for FileStorage {
     }
 }
 
-pub trait Storage {}
+pub trait Storage {
+    fn get_username(&self) -> String;
+    fn get_password(&self) -> String;
+    fn get_token(&self) -> String;
+    fn get_course(&self) -> String;
+    fn get_task(&self) -> String;
+    fn get_language(&self) -> String;
+    fn get_option(&self) -> String;
+    fn get_file(&self) -> String;
+    fn set_username(&mut self, val: String);
+    fn set_password(&mut self, val: String);
+    fn set_token(&mut self, val: String);
+    fn set_course(&mut self, val: String);
+    fn set_task(&mut self, val: String);
+    fn set_language(&mut self, val: String);
+    fn set_option(&mut self, val: String);
+    fn set_file(&mut self, val: String);
+    fn save(&self);
+    fn delete(&self);
+}
 
-impl Storage for FileStorage {}
+impl Storage for FileStorage {
+    fn get_username(&self) -> String {
+        self.username.to_string()
+    }
+    fn get_password(&self) -> String {
+        self.password.to_string()
+    }
+    fn get_token(&self) -> String {
+        self.token.to_string()
+    }
+    fn get_course(&self) -> String {
+        self.course.to_string()
+    }
+    fn get_task(&self) -> String {
+        self.task.to_string()
+    }
+    fn get_language(&self) -> String {
+        self.language.to_string()
+    }
+    fn get_option(&self) -> String {
+        self.option.to_string()
+    }
+    fn get_file(&self) -> String {
+        self.file.to_string()
+    }
+    fn set_username(&mut self, val: String) {
+        self.username = val;
+    }
+    fn set_password(&mut self, val: String) {
+        self.password = val;
+    }
+    fn set_token(&mut self, val: String) {
+        self.token = val;
+    }
+    fn set_course(&mut self, val: String) {
+        self.course = val;
+    }
+    fn set_task(&mut self, val: String) {
+        self.task = val;
+    }
+    fn set_language(&mut self, val: String) {
+        self.language = val;
+    }
+    fn set_option(&mut self, val: String) {
+        self.option = val;
+    }
+    fn set_file(&mut self, val: String) {
+        self.file = val;
+    }
+    fn save(&self) {
+        fs::write(FILENAME, json::to_string(&self)).unwrap();
+    }
+    fn delete(&self) {
+        fs::remove_file(FILENAME).unwrap();
+    }
```

Comment:
Instead of using `unwrap`, we should pass the error up in the return value with `Result`.

So, put `use anyhow::Result;` on the top of the file and:

```rust
fn delete(&self) -> Result<()> {
    fs::remove_file(FILENAME)?;
}
```